test/unittest/unittest_nntrainer_tensor.cpp

```diff
@@ -1210,6 +1939,88 @@ TEST(nntrainer_Tensor, print_large_size) {
   EXPECT_EQ(ss.str(), expected.str());
 }
 
+TEST(nntrainer_Tensor, external_loop_n) {
+  nntrainer::Tensor t = ranged(3, 5, 1, 4);
+  nntrainer::Tensor b = ranged(1, 5, 1, 4);
+
+  auto vector_func = [](float *buf, int stride, int size) {
+    float *cur = buf;
+    std::cerr << "[ ";
+    for (int i = 0; i < size; ++i) {
+      std::cerr << *cur << ' ';
+      cur += stride;
+    }
+    std::cerr << "]\n";
+  };
+
+  nntrainer::ExternalLoopInfo e;
+  EXPECT_NO_THROW(e = t.computeExternalLoop(b));
+
+  float *buf = t.getData();
+  float *mbuf = b.getData();
+
+  auto &t_strides = t.getStrides();
+  unsigned int offset;
+  unsigned int m_offset;
+
+  if (e.buffer_axis == -1) {
+    vector_func(buf, t_strides[3], e.buffer_size);
+    vector_func(mbuf, e.strides[3], e.buffer_size);
+  } else {
+    for (unsigned int b = 0; b < t.batch(); ++b) {
+      if (e.buffer_axis == 0) {
+        offset = b * t_strides[0];
+        m_offset = b * e.strides[0];
+        std::cerr << "=====================\n";
```

Comment:
std::cerr -> ml_logw?

Reply:
This test is not automated. It checks if the given tensor is well broadcasted by printing it. Should I disable the test?

Reply:
`This test checks by printing it` -> you mean this test prints for us to check ?

Reply:
Yes. more for a debug & demonstration purpose. I could've deleted this test but left it on purpose because it is useful.

Reply:
then disabling this is fine i guess.

Reply:
Disabled